traffic_portal/app/src/modules/private/roles/users/index.js

```diff
@@ -30,8 +30,8 @@ module.exports = angular.module('trafficPortal.private.roles.users', [])
 							roles: function($stateParams, roleService) {
 								return roleService.getRoles({ id: $stateParams.roleId });
 							},
-							roleUsers: function($stateParams, userService) {
-								return userService.getUsers({ roleId: $stateParams.roleId });
+							roleUsers: function(roles, userService) {
+								return userService.getUsers({ role: roles[0].name });
```

Comment:
I know this should never happen in the context of a UI that a user must sign in to use, but it still scares me to see potentially uncaught `TypeError`s being calculated in every digest cycle.

Reply:
i don't follow

Reply:
```javascript
const roles = [];
console.log(roles[0].name); //TypeError: cannot read property 'name' of undefined
```

Reply:
roles should never be an empty array as it pulls a valid role id from the url. i.e. tp.example.net/#!/roles/4/users

however, yes, you could change that role id to 999 but....don't do that. :)

Reply:
i suppose extra code could be added to catch the error but seems like it would just be adding more code for a scenario that is super rare and has no real benefit.

Reply:
Well in a proper Angular test bed we'd be mocking up services like the API so it doesn't depend on an actual, running instance of TO. But since we know that this code is always running in a world where it's connected to a TO where it's authenticated and users must have roles, we do know that the list of all roles must not be empty.

So I guess if we have no immediate plans to remove the tests' dependency on TO (and I don't think we do) then we can let this go.